src/app/services/parser.service.ts

```diff
@@ -0,0 +1,175 @@
+import { Injectable } from '@angular/core';
+import {TrackPoint} from '../../track';
+import {getDistance} from 'ol/sphere';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ParserService {
+
+  flightDuration;
+  totalDistance;
+  e2eDistance;
+  startAltitude;
+  stopAltitude;
+  highestPoint;
+
+  constructor() {}
+
+  TurnPointsDetection(trackData, nPoints) {
```

Comment:
I know you spent a lot of time working on this algorithm, could you detail a bit how it works in comments?

Reply:
Sure, good idea, will do

Reply:
Done